File: java/happy number.java
private int getNumber(int n) {
        //String num = String.valueOf(n);
        int sum = 0;
        
        while(n !=0){
            
            sum += (n%10)*(n%10);
            n = n/10;
        }
        return sum;
    }
    public boolean isHappy(int n) {
        HashSet<Integer> m = new HashSet<Integer>();
        
        while (n!=1){
            if(m.contains(n)){
                return false;
            }    
            m.add(n);
            n = getNumber(n);
            
        }
        return true;
        
    }

File: python/Happy_number.py
def isHappy(self, n):
        """
        :type n: int
        :rtype: bool
        """
        dic = list()
        while n!=1:
            if n in dic:
                return False
            dic.append(n)
            n = self.getNumber(n)
        return True
        
    def getNumber(self, n):
        sum = 0
        while n != 0:
            sum = sum + (n%10) * (n%10)
            n /= 10 
        return sum
